app/Importer/AssetModelImporter.php

```diff
@@ -0,0 +1,174 @@
+            $editingAssetModel = true;
+        } else {
+            $this->log('No Matching Model, Create a new one');
+            $assetModel = new AssetModel();
+        }
+
+        // Pull the records from the CSV to determine their values
+        $this->item['name'] = trim($this->findCsvMatch($row, 'name'));
+        $this->item['category'] = trim($this->findCsvMatch($row, 'category'));
+        $this->item['manufacturer'] = trim($this->findCsvMatch($row, 'manufacturer'));
+        $this->item['min_amt'] = trim($this->findCsvMatch($row, 'min_amt'));
+        $this->item['model_number'] = trim($this->findCsvMatch($row, 'model_number'));
+        $this->item['eol'] = trim($this->findCsvMatch($row, 'eol'));
+        $this->item['notes'] = trim($this->findCsvMatch($row, 'notes'));
+        $this->item['fieldset'] = trim($this->findCsvMatch($row, 'fieldset'));
+        $this->item['depreciation'] = trim($this->findCsvMatch($row, 'depreciation'));
+        $this->item['requestable'] = trim(($this->fetchHumanBoolean($this->findCsvMatch($row, 'requestable'))) == 1) ? 1 : 0;
+
+        if (!empty($this->item['category'])) {
+            if ($category = $this->createOrFetchCategory($this->item['category'])) {
+                $this->item['category_id'] = $category;
+            }
+        }
+        if (!empty($this->item['manufacturer'])) {
+            if ($manufacturer = $this->createOrFetchManufacturer($this->item['manufacturer'])) {
+                $this->item['manufacturer_id'] = $manufacturer;
+            }
+        }
+
+        if (!empty($this->item['depreciation'])) {
+            if ($depreciation = $this->fetchDepreciation($this->item['depreciation'])) {
+                $this->item['depreciation_id'] = $depreciation;
+            }
+        }
+
+        if (!empty($this->item['fieldset'])) {
+            if ($fieldset = $this->createOrFetchCustomFieldset($this->item['fieldset'])) {
+                $this->item['fieldset_id'] = $fieldset;
+            }
+        }
+
+        Log::debug('Item array is: ');
+        Log::debug(print_r($this->item, true));
+
+
+        if ($editingAssetModel) {
+            Log::debug('Updating existing model');
+            $assetModel->update($this->sanitizeItemForUpdating($assetModel));
+        } else {
+            Log::debug('Creating model');
+            $assetModel->fill($this->sanitizeItemForStoring($assetModel));
+            $assetModel->created_by = auth()->id();
+        }
+
+        if ($assetModel->save()) {
+            $this->log('AssetModel '.$assetModel->name.' created or updated from CSV import');
+            return $assetModel;
+
+        } else {
+            $this->log($assetModel->getErrors()->first());
+            $this->addErrorToBag($assetModel,  $assetModel->getErrors()->keys()[0], $assetModel->getErrors()->first());
+            return $assetModel->getErrors();
+        }
+
+    }
+
+
+    /**
+     * Fetch an existing depreciation, or create new if it doesn't exist.
```

Comment:
Since we don't create a new one if it doesn't exist, this comment should probably change.